src/Functions.hpp

```diff
@@ -0,0 +1,135 @@
+#ifndef FUNCTIONS_H
+#define FUNCTIONS_H
+
+#include <deal.II/base/function.h>
+#include "define.h"
+
+using namespace dealii;
+using namespace std;
+
+#if DIM == 1
```

Comment:
Preprocessor directives of this kind can and should usually be replaced with more modern (and less error-prone alternatives) such as `if constexpr` statements or template specializations.